src/garage/torch/distributions/tanh_normal.py

```diff
@@ -0,0 +1,154 @@
+"""A Gaussian distribution with tanh transformation."""
+from collections import namedtuple
+
+import torch
+from torch.distributions import Distribution, MultivariateNormal
+
+
+class TanhNormal(Distribution):
+    # pylint: disable=abstract-method
+    """A gaussian distribution that has a tanh transformation applied to it.
+
+    Algorithms like SAC and Pearl use this transformed distribution.
+    It can be thought of as a distribution of X where
+        Y ~ N(mean, cov)
+        X ~ tanh(Y)
+
+    Args:
+        mean(torch.tensor): The mean of this distribution.
+        cov(torch.tensor): The covariance of this distribution.
+        epsilon(float): Coefficient of stability used when calculating
+                        the log likehilhood of a sample on a distribution.
+
+    """
+
+    def __init__(self, mean, cov, epsilon=1e-6):
+        self._normal = MultivariateNormal(mean, cov)
+        self._epsilon = epsilon
```

Comment:
AFAICT epsilon is only used by `log_prob`, so why not make it an optional parameter of that function rather than the class?